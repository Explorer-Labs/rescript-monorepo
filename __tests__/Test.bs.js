// Generated by ReScript, PLEASE EDIT WITH CARE

import * as Curry from "bs-platform/lib/es6/curry.js";
import * as React from "react";
import * as ReactTestRenderer from "react-test-renderer";
import * as Css$RescriptMonorepo from "../src/styles/Css.bs.js";
import * as Sum$RescriptMonorepo from "../src/Sum.bs.js";
import * as Jest$RescriptMonorepo from "../src/bindings/Jest.bs.js";
import * as Utils$RescriptMonorepo from "../src/utils/Utils.bs.js";

function drinkAll(callback, flavour) {
  if (flavour !== "octopus") {
    return flavour;
  } else {
    return Curry._1(callback, flavour);
  }
}

var globalValue1 = {
  contents: 0
};

describe("Snapshots", (function () {
        var styles = Css$RescriptMonorepo.declaration("float", "left");
        var Test = function (Props) {
          var children = Props.children;
          return React.createElement("h1", {
                      className: styles
                    }, children);
        };
        test("renders with correct styles", (function () {
                var tree = ReactTestRenderer.create(React.createElement(Test, {
                            children: Utils$RescriptMonorepo.str("hello world")
                          })).toJSON();
                expect(tree).toMatchSnapshot();
                
              }));
        
      }));

describe("Common Matchers", (function () {
        test("toBe", (function () {
                expect(Sum$RescriptMonorepo.sum(1, 2)).toBe(3);
                
              }));
        test("two plus two is four", (function () {
                expect(4).toBe(4);
                
              }));
        test("two plus two is not five", (function () {
                expect(4).not.toBe(5);
                
              }));
        test("object assignment using rescript records", (function () {
                var dataUpdate = {
                  num: 2,
                  name: "test"
                };
                var dataMutable = {
                  name: "Baby",
                  age: 5
                };
                dataMutable.age = dataMutable.age + 1 | 0;
                expect({
                        num: 1,
                        name: "test"
                      }).toEqual({
                      num: 1,
                      name: "test"
                    });
                expect(dataUpdate).toEqual({
                      num: 2,
                      name: "test"
                    });
                expect(dataMutable).toEqual({
                      name: "Baby",
                      age: 6
                    });
                
              }));
        test("null", (function () {
                var $$null = null;
                expect($$null).toBeDefined();
                expect(undefined).not.toBeDefined();
                expect($$null).toBeNull();
                expect(undefined).not.toBeNull();
                expect(undefined).toBeUndefined();
                expect($$null).not.toBeUndefined();
                expect(true).toBeTruthy();
                expect($$null).not.toBeTruthy();
                expect($$null).toBeFalsy();
                expect(true).not.toBeFalsy();
                
              }));
        test("zero", (function () {
                expect(0).not.toBeNull();
                expect(0).toBeDefined();
                expect(0).not.toBeUndefined();
                expect(0).not.toBeTruthy();
                expect(0).toBeFalsy();
                
              }));
        test("two plus two", (function () {
                expect(4).toBeGreaterThan(3);
                expect(4).toBeGreaterThanOrEqual(3.5);
                expect(4).toBeLessThan(5);
                expect(4).toBeLessThanOrEqual(4.5);
                expect(4).toBe(4);
                expect(4).toEqual(4);
                
              }));
        test("adding floating point numbers", (function () {
                var value = 0.1 + 0.2;
                expect(value).toBeCloseTo(0.3);
                
              }));
        test("the shopping list has beer on it", (function () {
                var shoppingList = [
                  "diapers",
                  "kleenex",
                  "trash bags",
                  "paper towels",
                  "beer"
                ];
                expect(shoppingList).toContain("beer");
                expect(shoppingList).not.toContain("BEER");
                
              }));
        test("there is no I in team", (function () {
                expect("team").not.toMatch(/I/);
                
              }));
        test("but there is a 'stop' in Christoph", (function () {
                expect("Christoph").toMatch(/stop/);
                
              }));
        test("id should match object", (function () {
                var obj = {
                  id: "111",
                  product: "Test",
                  url: "https://jestjs.io/"
                };
                expect(obj).toEqual(expect.objectContaining({
                          id: "111"
                        }));
                expect(obj).toEqual(expect.not.objectContaining({
                          title: "Test"
                        }));
                
              }));
        test("String containing", (function () {
                expect("abcdefghijk").toEqual(expect.stringContaining("abcde"));
                expect("abcdefghijk").not.toEqual(expect.stringContaining("abce"));
                
              }));
        test("String matching", (function () {
                expect("abcdefghijk").toEqual(expect.stringContaining("abcde"));
                expect("abcdefghijk").not.toEqual(expect.stringContaining("abce"));
                
              }));
        
      }));

describe("Array Containing", (function () {
        var expected = [
          "Alice",
          "Bob"
        ];
        Jest$RescriptMonorepo.it("matches even if received contains additional elements", (function () {
                expect([
                        "Alice",
                        "Bob",
                        "Eve"
                      ]).toEqual(expect.arrayContaining(expected));
                
              }));
        Jest$RescriptMonorepo.it("matches even if received contains additional elements", (function () {
                expect([
                        "Bob",
                        "Eve"
                      ]).toEqual(expect.not.arrayContaining(expected));
                
              }));
        Jest$RescriptMonorepo.it("does not match if received does not contain expected elements", (function () {
                expect([
                        "Bob",
                        "Eve"
                      ]).not.toEqual(expect.arrayContaining(expected));
                
              }));
        
      }));

describe("stringMatching in arrayContaining", (function () {
        var expected = [
          expect.stringMatching(/^Alic/),
          expect.stringMatching(/^[BR]ob/)
        ];
        Jest$RescriptMonorepo.it("matches even if received contains additional elements", (function () {
                expect([
                        "Alicia",
                        "Roberto",
                        "Evelina"
                      ]).toEqual(expect.arrayContaining(expected));
                
              }));
        Jest$RescriptMonorepo.it("does not match if received does not contain expected elements", (function () {
                expect([
                        "Roberto",
                        "Evelina"
                      ]).not.toEqual(expect.arrayContaining(expected));
                
              }));
        
      }));

describe("Mock function", (function () {
        test("drinks something lemon-flavoured", (function () {
                var mockFn = jest.fn();
                Curry._1(mockFn, undefined);
                return Jest$RescriptMonorepo.toHaveBeenCalled(expect(mockFn), undefined);
              }));
        
      }));

describe("Mock function with implementation", (function () {
        test("drinks something lemon-flavoured", (function () {
                var returnsTrue = jest.fn(function (param) {
                      return true;
                    });
                expect(Curry._1(returnsTrue, undefined)).toBe(true);
                
              }));
        
      }));

describe("Lifecycle", (function () {
        var persistentValue = {
          contents: 0
        };
        beforeEach(function () {
              persistentValue.contents = persistentValue.contents + 1 | 0;
              
            });
        afterEach(function () {
              persistentValue.contents = persistentValue.contents - 1 | 0;
              
            });
        test("runs lifecycle the first time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        test("runs lifecycle each time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        
      }));

describe("Skips other tests", (function () {
        var persistentValue = {
          contents: 0
        };
        beforeEach(function () {
              persistentValue.contents = persistentValue.contents + 1 | 0;
              
            });
        afterEach(function () {
              persistentValue.contents = persistentValue.contents - 1 | 0;
              
            });
        test.skip("runs lifecycle the first time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        test("runs lifecycle each time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        
      }));

describe.skip("Lifecycle", (function () {
        var persistentValue = {
          contents: 0
        };
        beforeEach(function () {
              persistentValue.contents = persistentValue.contents + 1 | 0;
              
            });
        afterEach(function () {
              persistentValue.contents = persistentValue.contents - 1 | 0;
              
            });
        test("runs lifecycle the first time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        test("runs lifecycle each time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        
      }));

describe("Test each functionality", (function () {
        test.each([
                [
                  1,
                  1,
                  2
                ],
                [
                  1,
                  2,
                  3
                ],
                [
                  2,
                  1,
                  3
                ]
              ])(".add(%i, %i)", (function (a, b, expected) {
                expect(a + b | 0).toBe(expected);
                
              }));
        Jest$RescriptMonorepo.itEach([
                [
                  1,
                  1,
                  2
                ],
                [
                  1,
                  2,
                  3
                ],
                [
                  2,
                  1,
                  3
                ]
              ])(".add(%i, %i)", (function (a, b, expected) {
                expect(a + b | 0).toBe(expected);
                
              }));
        
      }));

describe("Test skip each", (function () {
        test.skip.each([
                [
                  1,
                  1,
                  2
                ],
                [
                  1,
                  2,
                  3
                ],
                [
                  2,
                  1,
                  3
                ]
              ])(".add(%i, %i)", (function (a, b, expected) {
                expect(a + b | 0).toBe(expected);
                
              }));
        
      }));

describe("Todo works", (function () {
        test.todo("Write test 1");
        test.todo("Write test 2");
        test.todo("Write test 3");
        
      }));

describe.each([
        [
          1,
          1,
          2
        ],
        [
          1,
          2,
          3
        ],
        [
          2,
          1,
          3
        ]
      ])(".add(%i, %i)", (function (a, b, expected) {
        test("returns " + String(expected), (function () {
                expect(a + b | 0).toBe(expected);
                
              }));
        test("returned value not be greater than " + String(expected), (function () {
                expect(a + b | 0).not.toBeGreaterThan(expected);
                
              }));
        test("returned value not be less than " + String(expected), (function () {
                expect(a + b | 0).not.toBeLessThan(expected);
                
              }));
        
      }));

var sum = Sum$RescriptMonorepo.sum;

export {
  sum ,
  drinkAll ,
  globalValue1 ,
  
}
/*  Not a pure module */
