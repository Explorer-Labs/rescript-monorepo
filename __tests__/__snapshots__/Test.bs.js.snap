// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders with correct styles 1`] = `
<h1
  className="float: left;"
>
  hello world
</h1>
`;
